### DISCUSSION TITLE
Key Value from Kafka

### DISCUSSION THREAD

**Discussion by gragha** [edenhill/librdkafka#3407] [Q&A]
Hi. We are writing into Kafka queue from a vended product using the librdkafka.dll. We have a requirement where we need to set the key value from part of the actual JSON message we write into Kafka. Unfortunately the Vended tool allows us to only specify properties which means I can only use the configuration properties given here. I am not able to find any equivalent command to parse the text and get key value from the text and pass it on along with the message. Key becomes important to us due to maintaining sequencing in the messages. 

Any help on this regard is greatly appreciated.

**Comment by rolandyoung** [comment]
Does the vended product not provide any mechanism for you to supply the key? librdkafka expects the key and message to be supplied separately from each other - there is no mechanism to set one from the other.
